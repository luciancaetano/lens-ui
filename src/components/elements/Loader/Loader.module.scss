@import '../../../style-lib.scss';


.loader {
  @include lens-ui-typography;
  display: flex;
  overflow: hidden;
  font-size: .65625rem;
  border-radius: .25rem;
  position: relative;

  &__content {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 1;
    display: flex;
    align-items: center;
    justify-content: center;
  }
}