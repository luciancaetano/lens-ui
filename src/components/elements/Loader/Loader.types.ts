import React from 'react';
import {
  IntentType, ITestableProps,
} from '../../../types';

export interface ILoaderProps extends ITestableProps, React.HtmlHTMLAttributes<HTMLElement> {
  type?: 'eclipse' | 'spinner' | 'oval';
  intent?: IntentType;
  size?: number;
}
