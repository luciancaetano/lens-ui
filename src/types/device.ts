export type DeviceOrientationType = 'portrait' | 'landscape';
export type DeviceOSType = 'android' | 'ios' | false;

export interface IDeviceWindow {
  width: number;
  height: number;
}

export interface IDeviceContextType {
  windowSize: IDeviceWindow;
  orientation: DeviceOrientationType;
  online: boolean;
  isUsingDeviceProvider: boolean;
  darkMode: boolean;
  lg: boolean;
  md: boolean;
  sm: boolean;
  xs: boolean;
}

export interface IDeviceInfo {
  windowSize: IDeviceWindow;
  orientation: DeviceOrientationType;
  os: DeviceOSType;
  online: boolean;
  darkMode: boolean;
  lg: boolean;
  md: boolean;
  sm: boolean;
  xs: boolean;
}
