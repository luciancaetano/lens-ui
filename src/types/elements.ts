export interface ITestableProps {
  testingID?: string;
}
