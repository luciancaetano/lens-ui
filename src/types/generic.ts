export const Intents = {
  primary: 'primary',
  secondary: 'secondary',
  success: 'success',
  info: 'info',
  warning: 'warning',
  danger: 'danger',
};

export type IntentType = keyof typeof Intents | null;

export const AdvancedIntent = {
  ...Intents,
  light: 'light',
  dark: 'dark',
};

export type AdvancedIntentType = keyof typeof AdvancedIntent;

export type UserScriptStateType = 'loading' | 'idle' | 'ready' | 'error';
