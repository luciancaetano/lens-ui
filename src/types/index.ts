export * from './elements';
export * from './generic';
export * from './device';
