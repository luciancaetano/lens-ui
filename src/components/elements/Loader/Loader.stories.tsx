/* eslint-disable react/destructuring-assignment */
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Loader from './Loader';
import Icon from '../Icon/Icon';
import LensProvider from '../../providers/LensProvider/LensProvider';
import '../../../styles';

export default {
  title: '2. Components/Loader',
  component: Loader,
  decorators: [
    (Story) => <LensProvider>{Story()}</LensProvider>,
  ],
} as ComponentMeta<typeof Loader>;

const LoaderTemplate: ComponentStory<typeof Loader> = (args) => (
  <Loader {...args}>
    <Icon name="BsCamera" size={30} />
  </Loader>
);

export const _Loader = LoaderTemplate.bind({});
_Loader.args = {
  type: 'eclipse',
  intent: 'primary',
  size: 5,
};
