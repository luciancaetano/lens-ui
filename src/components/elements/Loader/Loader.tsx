import React from 'react';
import clsx from 'clsx';
import styles from './Loader.module.scss';
import { ILoaderProps } from './Loader.types';
import { CLASSES } from '../../../css-classes';

const Loader: React.FC<ILoaderProps> = ({
  className, testingID, id, children, type = 'eclipse', intent = 'primary', size = 5,
}) => (
  <div
    id={id}
    data-testid={testingID}
    className={clsx(styles.loader, CLASSES.ComponentName('Loader'), className)}
    style={{ width: `${size}rem`, height: `${size}rem` }}
  >
    <div className={styles.loaderContent}>
      {children}
    </div>
    {type === 'oval' && (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="var(--loader-width)"
        height="var(--loader-width)"
        stroke={`var(--lens-ui-intents-${intent})`}
        viewBox="-5.8 -5.8 50 50"
      >
        <g
          fill="none"
          fillRule="evenodd"
          strokeWidth="2"
          transform="translate(1 1)"
        >
          <circle cx="18" cy="18" r="18" strokeOpacity="0.5" />
          <path d="M36 18c0-9.94-8.06-18-18-18">
            <animateTransform
              attributeName="transform"
              dur="1s"
              from="0 18 18"
              repeatCount="indefinite"
              to="360 18 18"
              type="rotate"
            />
          </path>
        </g>
      </svg>
    )}
    {type === 'eclipse' && (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        style={{ margin: 'auto' }}
        width="var(--loader-width)"
        height="var(--loader-width)"
        display="block"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 100 100"
      >
        <path fill={`var(--lens-ui-intents-${intent})`} d="M10 50a40 40 0 0080 0 40 42 0 01-80 0">
          <animateTransform
            attributeName="transform"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            type="rotate"
            values="0 50 51;360 50 51"
          />
        </path>
      </svg>
    )}
    {type === 'spinner' && (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        style={{ margin: 'auto' }}
        width="var(--loader-width)"
        height="var(--loader-width)"
        display="block"
        preserveAspectRatio="xMidYMid"
        viewBox="20 20 60 60"
      >
        <rect width="6" height="12" x="47" y="24" fill={`var(--lens-ui-intents-${intent})`} rx="3" ry="6">
          <animate
            attributeName="opacity"
            begin="-0.9166666666666666s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(30 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.8333333333333334s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(60 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.75s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(90 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.6666666666666666s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(120 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.5833333333333334s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(150 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.5s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(180 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.4166666666666667s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(210 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.3333333333333333s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(240 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.25s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(270 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.16666666666666666s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(300 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="-0.08333333333333333s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
        <rect
          width="6"
          height="12"
          x="47"
          y="24"
          fill={`var(--lens-ui-intents-${intent})`}
          rx="3"
          ry="6"
          transform="rotate(330 50 50)"
        >
          <animate
            attributeName="opacity"
            begin="0s"
            dur="1s"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="1;0"
          />
        </rect>
      </svg>
    )}
  </div>
);

export default Loader;
