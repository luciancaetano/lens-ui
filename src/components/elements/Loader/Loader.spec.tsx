import React from 'react';

import { render } from '@testing-library/react';
import { IntentEnum } from '../../../types';
import Loader from './Loader';

describe('<Loader/>', () => {
  it('should render correct Loader children', () => {
    const childrenText = 'hello_world';
    const { getByText } = render(<Loader>{childrenText}</Loader>);

    expect(getByText(childrenText)).toBeInTheDocument();
  });

  it('should render correct Loader intents', () => {
    const testingId = 'my-testing-id';
    const { getByTestId, rerender } = render(<Loader testingID={testingId} intent={null}>1</Loader>);

    Object.keys(IntentEnum).forEach((intent) => {
      rerender(<Loader testingID={testingId} intent={intent as any} size={20}>1</Loader>);
      expect(getByTestId(testingId)).toMatchSnapshot(intent);
    });
  });

  it('should render correct Loader types', () => {
    const testingId = 'my-testing-id';
    const { getByTestId, rerender } = render(<Loader testingID={testingId} intent={null}>1</Loader>);

    ['eclipse', 'spinner', 'oval'].forEach((type) => {
      rerender(<Loader testingID={testingId} type={type as any}>1</Loader>);
      expect(getByTestId(testingId)).toMatchSnapshot(type);
    });
  });
});
