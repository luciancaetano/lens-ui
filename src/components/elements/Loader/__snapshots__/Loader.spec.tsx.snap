// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader/> should render correct Loader intents: danger 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="danger"
  data-testid="my-testing-id"
  style="width: 20rem; height: 20rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="0 0 100 100"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 50a40 40 0 0080 0 40 42 0 01-80 0"
      fill="var(--lens-ui-intents-danger)"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 51;360 50 51"
      />
    </path>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader intents: info 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="info"
  data-testid="my-testing-id"
  style="width: 20rem; height: 20rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="0 0 100 100"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 50a40 40 0 0080 0 40 42 0 01-80 0"
      fill="var(--lens-ui-intents-info)"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 51;360 50 51"
      />
    </path>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader intents: primary 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="primary"
  data-testid="my-testing-id"
  style="width: 20rem; height: 20rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="0 0 100 100"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 50a40 40 0 0080 0 40 42 0 01-80 0"
      fill="var(--lens-ui-intents-primary)"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 51;360 50 51"
      />
    </path>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader intents: secondary 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="secondary"
  data-testid="my-testing-id"
  style="width: 20rem; height: 20rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="0 0 100 100"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 50a40 40 0 0080 0 40 42 0 01-80 0"
      fill="var(--lens-ui-intents-secondary)"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 51;360 50 51"
      />
    </path>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader intents: success 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="success"
  data-testid="my-testing-id"
  style="width: 20rem; height: 20rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="0 0 100 100"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 50a40 40 0 0080 0 40 42 0 01-80 0"
      fill="var(--lens-ui-intents-success)"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 51;360 50 51"
      />
    </path>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader intents: warning 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="warning"
  data-testid="my-testing-id"
  style="width: 20rem; height: 20rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="0 0 100 100"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 50a40 40 0 0080 0 40 42 0 01-80 0"
      fill="var(--lens-ui-intents-warning)"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 51;360 50 51"
      />
    </path>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader types: eclipse 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="primary"
  data-testid="my-testing-id"
  style="width: 5rem; height: 5rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="0 0 100 100"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M10 50a40 40 0 0080 0 40 42 0 01-80 0"
      fill="var(--lens-ui-intents-primary)"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 51;360 50 51"
      />
    </path>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader types: oval 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="primary"
  data-testid="my-testing-id"
  style="width: 5rem; height: 5rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    height="var(--loader-width)"
    stroke="var(--lens-ui-intents-primary)"
    viewBox="-5.8 -5.8 50 50"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
      fill-rule="evenodd"
      stroke-width="2"
      transform="translate(1 1)"
    >
      <circle
        cx="18"
        cy="18"
        r="18"
        stroke-opacity="0.5"
      />
      <path
        d="M36 18c0-9.94-8.06-18-18-18"
      >
        <animatetransform
          attributeName="transform"
          dur="1s"
          from="0 18 18"
          repeatCount="indefinite"
          to="360 18 18"
          type="rotate"
        />
      </path>
    </g>
  </svg>
</div>
`;

exports[`<Loader/> should render correct Loader types: spinner 1`] = `
<div
  class="loader"
  data-lens-element="loader"
  data-lens-intent="primary"
  data-testid="my-testing-id"
  style="width: 5rem; height: 5rem;"
>
  <div
    class="loaderContent"
  >
    1
  </div>
  <svg
    display="block"
    height="var(--loader-width)"
    preserveAspectRatio="xMidYMid"
    style="margin: auto;"
    viewBox="20 20 60 60"
    width="var(--loader-width)"
    xmlns="http://www.w3.org/2000/svg"
  >
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.9166666666666666s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(30 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.8333333333333334s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(60 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.75s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(90 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.6666666666666666s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(120 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.5833333333333334s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(150 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.5s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(180 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.4166666666666667s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(210 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.3333333333333333s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(240 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.25s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(270 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.16666666666666666s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(300 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="-0.08333333333333333s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
    <rect
      fill="var(--lens-ui-intents-primary)"
      height="12"
      rx="3"
      ry="6"
      transform="rotate(330 50 50)"
      width="6"
      x="47"
      y="24"
    >
      <animate
        attributeName="opacity"
        begin="0s"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        values="1;0"
      />
    </rect>
  </svg>
</div>
`;
